pkg/apis/v1beta1/raft.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RaftStorageClassGroup raft storage class group
+const RaftStorageClassGroup = "storage.cloud.atomix.io"
+
+// RaftStorageClassVersion raft storage class version
+const RaftStorageClassVersion = "v1beta1"
+
+// RaftStorageClassKind raft storage class kind
+const RaftStorageClassKind = "RaftStorageClass"
+
+// RaftStorageClassSpec defines the desired state of RaftStorageClass
+type RaftStorageClassSpec struct {
```

Comment:
We can forgo this for the first implementation, but we're definitely going to need to support a `StorageClass` here or something like `StatefulSet`'s `volumeClaimTemplates`. Users should be able to dictate how data is stored in the Raft storage class since Raft is persistent.

Reply:
Yeah that was in my list of things to do in the next round when we have something ready.